documentation/src/main/asciidoc/topics/cache_api.adoc

```diff
@@ -164,7 +164,7 @@ include::code_examples/ListenerMyNonBlocking.java[]
 ----
 
 ==== Asynchronous thread pool
-To tune the thread pool used to dispatch such asynchronous notifications, use the link:http://docs.jboss.org/infinispan/{infinispanversion}/configdocs/infinispan-config-{infinispanversion}.html[`<listener-executor />`] XML element in your configuration file.
+To tune the thread pool used to dispatch such asynchronous notifications, use the link:{configdocroot}/infinispan-config-{infinispanversion}.html[`<listener-executor />`] XML element in your configuration file.
 
 [[cache_asynchronous_api]]
 ==  Asynchronous API
```

Comment:
links need to be updated in this subsection. 
Also, can you double-check the text? We no longer return a Future but a CompletableFuture.